codespan/src/span.rs

```diff
@@ -1,257 +1,131 @@
 #[cfg(feature = "serialization")]
 use serde::{Deserialize, Serialize};
-use std::{cmp, cmp::Ordering, fmt};
+use std::fmt;
+use std::ops::Range;
 
-use crate::index::{ByteIndex, Index};
+use crate::ByteIndex;
 
-/// A region of code in a source file
-#[derive(Debug, Clone, Copy, Default, PartialEq, Eq, Ord, PartialOrd)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 #[cfg_attr(feature = "serialization", derive(Deserialize, Serialize))]
 #[cfg_attr(feature = "memory_usage", derive(heapsize_derive::HeapSizeOf))]
-pub struct Span<I> {
```

Comment:
I actually used `Span` with my own position type :/. Arguably that isn't what it was meant for however.

Reply:
Ahh, interesting! Yeah, I wasn't sure how useful that was. I could add it back in though.